src/valkeycluster.c

```diff
@@ -3687,6 +3687,20 @@ void valkeyClusterAsyncFree(valkeyClusterAsyncContext *acc) {
     vk_free(acc);
 }
 
+struct valkeyClusterNodeIterator {
+    valkeyClusterContext *cc;
+    uint64_t route_version;
+    int retries_left;
+    dictIterator di;
+};
+
+/* Get an initated iterator for iterating over current cluster nodes */
+valkeyClusterNodeIterator *valkeyClusterGetNodeIterator(valkeyClusterContext *cc) {
```

Comment:
Shall we call this create instead of get? A getter doesn't normally allocate anything.

```suggestion
valkeyClusterNodeIterator *valkeyClusterCreateNodeIterator(valkeyClusterContext *cc) {
```